src/app/loadout-drawer/LoadoutDrawerContainer.tsx

```diff
@@ -0,0 +1,177 @@
+
+  // Only react to add item if there's not a loadout open (otherwise it'll be handled in the loadout drawer!)
+  useEventBusListener(
+    addItem$,
+    useCallback(
+      (item: DimItem) => {
+        if (!hasInitialLoadout) {
+          // If we don't have a loadout, this action was invoked via the "+ Loadout" button in item actions
+          let owner: DimStore =
+            item.owner === 'vault' ? getCurrentStore(stores)! : getStore(stores, item.owner)!;
+
+          if (item.classType !== DestinyClass.Unknown && item.classType !== owner.classType) {
+            const matchingStore = stores.find((s) => s.classType === item.classType);
+            if (!matchingStore) {
+              showNotification({
+                type: 'warning',
+                title: t('Loadouts.ClassTypeMissing', { className: item.classTypeNameLocalized }),
+              });
+              return;
+            }
+            owner = matchingStore;
+          }
+
+          const classType =
+            item.classType === DestinyClass.Unknown ? owner.classType : item.classType;
+          const draftLoadout = newLoadout('', [], classType);
+          draftLoadout.items.push({
+            id: item.id,
+            hash: item.hash,
+            equip: true,
+            amount: item.amount ?? 1,
+          });
+          setInitialLoadout({
+            loadout: draftLoadout,
+            storeId: owner.id,
+            isNew: true,
+            showClass: true,
+          });
+        }
+      },
+      [hasInitialLoadout, stores]
+    )
+  );
+
+  // Load in a full loadout specified in the URL
+  useEffect(() => {
+    if (!stores.length || !defs?.isDestiny2()) {
+      return;
+    }
+    const searchParams = new URLSearchParams(queryString);
+    const loadoutJSON = searchParams.get('loadout');
+    if (loadoutJSON) {
+      try {
+        const parsedLoadout = convertDimApiLoadoutToLoadout(JSON.parse(loadoutJSON));
+        if (parsedLoadout) {
+          const storeId =
+            parsedLoadout.classType === DestinyClass.Unknown
+              ? getCurrentStore(stores)?.id
+              : stores.find((s) => s.classType === parsedLoadout.classType)?.id;
+
+          if (!storeId) {
+            warnMissingClass(parsedLoadout.classType, defs);
+            return;
+          }
+
+          parsedLoadout.id = uuidv4();
+          parsedLoadout.items = parsedLoadout.items.map((item) => ({
+            ...item,
+            id:
+              item.id === '0'
+                ? // We don't save consumables in D2 loadouts, but we may omit ids in shared loadouts
+                  // (because they'll never match someone else's inventory). So
+                  // instead, pick a random ID. It's possible these will
+                  // conflict with something already in the user's inventory but
+                  // it's not likely.
+                  Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString()
```

Comment:
I know this code was only moved but do item IDs have to be numeric? Vendor items already use the form `vendorHash-vendorItemIndex`, couldn't this just be `loadout-0`, `loadout-1` etc.?

Reply:
I suppose they don't! Prefixing them like that is a good idea.